lightning/src/ln/outbound_payment.rs

```diff
@@ -54,6 +57,8 @@ pub(crate) enum PendingOutboundPayment {
 		expiration: StaleExpiration,
 		retry_strategy: Retry,
 		max_total_routing_fee_msat: Option<u64>,
+		invoice_request: Option<InvoiceRequest>,
+		context: OffersContext,
```

Comment:
Let's just pass the `Nonce`. We can re-create `OffersContext::OutboundPayment` from it, which also prevents someone from inadvertently passing a different variant. The tests otherwise would also need `OffersContext::OutboundPayment` since `OffersContext::Unknown` is being removed in another PR.